src/Chapter2.hs

```diff
@@ -332,11 +332,17 @@ yummy functions.
 
 Don't forget that you can load this module in GHCi to call functions
 from it!
+    
+3 5 0,1,2,3,4,5,6,7
 
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList a b l
+ | a < 0 || b < 0 = []
+ | a > b = []
+ | (length l) < a = []
```

Comment:
That is a nice catch!
However, this is now free operation. This creates an additional walk through each element of the list, as `length` works this way. `length` for lists is not instant and cheap, so that you need to get through each element to count the size of the list.

with the following case you have, you can cover this case as well, `drop a l` will leave you with the empty list, so the case of `length l < a` is covered in there.

Reply:
Thanks for this suggestion, not that you explained, I got it!